logs/field.go

```diff
@@ -0,0 +1,211 @@
+package logs
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/ydb-platform/ydb-go-sdk/v3/trace"
+)
+
+// Field represents typed log field (a key-value pair). Adapters should determine
+// Field's type based on Type and use the corresponding getter method to retrieve
+// the value:
+//
+//	switch f.Type {
+//	case logs.IntType:
+//	    var i int = f.Int()
+//	    // handle int value
+//	case logs.StringType:
+//	    var s string = f.String()
+//	    // handle string value
+//	//...
+//	}
+//
+// Getter methods must not be called on fields with wrong Type (e.g. calling String()
+// on fields with Type != StringType).
+type Field struct {
+	Type FieldType
+	Key  string
+
+	vint int64
+	vstr string
+	vany interface{}
+}
+
+func String(key string, value string) Field {
+	return Field{
+		Type: StringType,
+		Key:  key,
+		vstr: value,
+	}
+}
+
+func (f Field) String() string {
```

Comment:
Maybe getters in Field should be called `GetX` instead of `X`? E.g.
```go
func (f Field) GetString() string {...}
```
This will make it easier to specify the purpose of such methods. Also, it will remove unwanted associations: now type `Field` implements `fmt.Stringer`, but semantically `func (Field) String() string` has a very different meaning.

Reply:
Get* is a java and c# like naming. In go it is unusual naming as I know
zapcore.Field have public fields. This is unsafed way, because fields are have read write access.
Why need use String() as fmt.Stringer interface? In which cases?

Reply:
Yes, @rekby is right about what I mean. I like the idea of renaming `f.String() => f.StringValue()` because `GetString` really does not look good in Go.

Reply:
Need final decision on this naming issue. `f.Foo() -> f.FooValue()` & `f.Fallback() -> f.String()`?

Reply:
I think f.Fallback() -> f.String() is good and I like f.FooValue() for get values.  

fmt.Stringer is very userful interface and often it used implicity, it is very easy to write fmt.Println(f) and it should work good by my opinion.